### DISCUSSION TITLE
How can I enable a repo to be used as a prompt?

### DISCUSSION THREAD

**Discussion by TheOnlyTails** [marlonrichert/zsh-snap#168] [Q&A] [answered]
I'm working with @JanDeDobbeleer on integrating https://github.com/JanDeDobbeleer/oh-my-posh with Znap (https://github.com/JanDeDobbeleer/oh-my-posh/issues/1881), and I was looking around on how to enable something like `znap prompt JanDeDobbeleer/oh-my-posh`, and have so far found nothing.
Are there certain files that need to be created or something?

**Comment by marlonrichert** [comment]
The first time I read about Oh-My-Posh. I have no idea how it works. 

How do you normally start your prompt with Oh-My-Posh?

**Reply by TheOnlyTails** [reply]
```zsh
eval "$(oh-my-posh prompt init zsh)"
```
[docs](https://ohmyposh.dev/docs/prompt)

**Comment by marlonrichert** [comment, marked as answer]
> How do you normally start your prompt with Oh-My-Posh?

> ```zsh
> eval "$(oh-my-posh prompt init zsh)"
> ```


Then this should work:
```zsh
znap eval oh-my-posh 'oh-my-posh prompt init zsh'
znap prompt
```

**Reply by TheOnlyTails** [reply]
I know, but I'd like to init it with the github repo:
```zsh
znap prompt jandedobbeleer/oh-my-posh
```

**Reply by JanDeDobbeleer** [reply]
@TheOnlyTails nobody seems to answer that question and the chance is highly likely that this is only possible for native zsh plugins.

**Reply by marlonrichert** [reply]
That’s not possible, because the repo contains neither a plugin file nor a theme file.

**Reply by TheOnlyTails** [reply]
I'm sure @JanDeDobbeleer would be willing to help with whatever's necessary (like a plugin file).

**Reply by marlonrichert** [reply]
@JanDeDobbeleer This is Znap’s code for finding prompt themes: https://github.com/marlonrichert/zsh-snap/blob/main/functions/.znap.prompt.load

Just add one of the files that it looks for and it will load it.

**Reply by JanDeDobbeleer** [reply]
@marlonrichert thanks. That does mean it won't work for oh-my-posh this way as it requires a working zsh script to be loaded which we don't have.

**Reply by marlonrichert** [reply]
@JanDeDobbeleer All you need to put into that file is `eval "$(oh-my-posh prompt init zsh)"`

**Reply by JanDeDobbeleer** [reply]
@marlonrichert that would imply we need to add the download there as well as it's a bit silly no? It swaps one line with another from a `.zshrc` perspective.

**Reply by marlonrichert** [reply]
Yeah, I don’t see any point to it either. 🙂
